File: python/2501.LongestSquareStreakinanArray.py
from collections import Counter
from typing import List


class Solution:
    def longestSquareStreak(self, nums: List[int]) -> int:
        """
        Finds the longest square streak in an array.

        Args:
          nums (List[int]): The input array of numbers.

        Returns:
          int: The length of the longest square streak, or -1 if there is no streak.
        """
        # Create a counter to store the count of each number's square
        count: Counter[int, int] = Counter()
        # Iterate through the sorted numbers and update the count of each number's square
        for x in sorted(nums):
            count[x * x] = count[x] + 1
        # Find the maximum count of any number's square
        res: int = max(count.values())
        # Return the result, or -1 if the maximum count is less than 2
        return res if res >= 2 else -1

File: typescript/medium/2501-longest-square-streak-in-an-array.ts
/**
 * Finds the longest square streak in an array.
 * 
 * @param {number[]} nums - The input array of numbers.
 * @returns {number} - The length of the longest square streak, or -1 if there is no streak.
 */
export function longestSquareStreak(nums: number[]): number {
  // Create a map to store the count of each number's square
  const count: Map<number, number> = new Map();
  // Iterate through the sorted numbers and update the count of each number's square
  for (const x of nums.sort((a, b) => a - b)) {
    count.set(x * x, (count.get(x) || 0) + 1);
  }
  // Find the maximum count of any number's square
  const res: number = Math.max(...Array.from(count.values()));
  // Return the result, or -1 if the maximum count is less than 2
  return res >= 2 ? res : -1;
};